### PR TITLE
Add support for the STM32F429ZI target

### PR DESCRIPTION
**Pull request by AlessandroA**
Putting this up to test the CI with the ST target.

@Patater @niklas-arm

### CODE REVIEW

**Comment by Patater** [comment]
CI failed because DISCO is a little bit too slow.

```
12:12:39 [1495537942.33][CONN][RXD] Main loop count: 31

12:12:39 [1495537942.86][CONN][RXD] Main loop count: 32

12:14:31 [1495538043.26][HTST][INF] test suite run finished after 120.25 sec...
12:14:31 [1495538043.26][HTST][ERR] Expected output [Main loop count: 33] not received in log.
```

**Review by Patater — COMMENTED** [review]
Consider moving the "Reduce memory footprint of this app" and "Add linker script memory filler for ARMv7-M MPU" commits to before adding support for new targets.

File: README.md [hunk 1]
```diff
@@ -4,9 +4,11 @@ This is a simple example to show how to write a uVisor-secured threaded applicat
...
 Supported devices:
 
-| Target | Toolchain | Baud rate |
-|--------|-----------|-----------|
-| `K64F` | `GCC_ARM` | 9600      |
+| Target            | Toolchain | Baud rate |
+|-------------------|-----------|-----------|
+| `K64F`            | `GCC_ARM` | 9600      |
+| `DISCO_F429ZI`    | `GCC_ARM` | 9600      |
+| `EFM32GG_STK3700` | `GCC_ARM` | 9600      |
```
Comment (Patater) [inline README.md:11]: Mention fixing https://github.com/ARMmbed/mbed-os-example-uvisor-thread/issues/33 in the commit message to auto-close the issue.

Comment (AlessandroA) [reply README.md:11]: Done.

**Review by Patater — APPROVED** [review]

**Comment by AlessandroA** [comment]
This can be merged once our mbed OS updates are merged into mbed OS.

**Comment by Patater** [comment]
retest uvisor

**Comment by AlessandroA** [comment]
retest uvisor

**Pull request merged**